diff --git a/dist/index.cjs b/dist/index.cjs
index 15c9728c6ad3eefa66306cfd3906c7c5311af06c..b3c3348f3effd53b6b18534a681dc289da5d56cd 100644
--- a/dist/index.cjs
+++ b/dist/index.cjs
@@ -163,7 +163,6 @@ var PureEditorContent = class extends react.default.Component {
 		editor.isEditorContentInitialized = false;
 		if (!editor.isDestroyed) {
 			editor.unregisterPlugin(mobileEnterPluginKey);
-			editor.view.setProps({ nodeViews: {} });
 		}
 		editor.contentComponent = null;
 		try {
diff --git a/dist/index.js b/dist/index.js
index 54376b761251f4618b2a8c17585824c8d2c1d64c..6dffb88c1fced6472bf33a2b58e7807303dab9d8 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -139,7 +139,6 @@ var PureEditorContent = class extends React.Component {
 		editor.isEditorContentInitialized = false;
 		if (!editor.isDestroyed) {
 			editor.unregisterPlugin(mobileEnterPluginKey);
-			editor.view.setProps({ nodeViews: {} });
 		}
 		editor.contentComponent = null;
 		try {
diff --git a/src/EditorContent.tsx b/src/EditorContent.tsx
index cedf56dba1088d0370a03cbe27b4457612a7634d..2693963a27333cd66af1a40269f5454e5db013cb 100644
--- a/src/EditorContent.tsx
+++ b/src/EditorContent.tsx
@@ -170,9 +170,6 @@ export class PureEditorContent extends React.Component<
 
     if (!editor.isDestroyed) {
       editor.unregisterPlugin(mobileEnterPluginKey)
-      editor.view.setProps({
-        nodeViews: {},
-      })
     }
 
     editor.contentComponent = null
